apis/url_test.go

```diff
@@ -71,6 +81,16 @@ func TestParseURL(t *testing.T) {
 				t.Fatalf("ParseURL() = %v, wanted error", got)
 			}
 
+			if tc.nonEmpty {
```

Comment:
can we just do `if tc.nonEmpty == got.IsEmpty()`, or which is better `nonEmpty -> wantEmpty` and then compare inequality?

Reply:
My thought process that led to this code:

1. When I started, there were more "empty" than non-empty cases, and I wanted the default to be simpler. At this point, there are more non-empty cases, so I can change this to `wantEmpty`.

   It also looks like the "empty" and "empty string" tests are the same, so I may remove one.

2. I started by writing this as `if tc.nonEmpty == got.IsEmpty()`, but then realized that I wanted different error messages for the two cases. I'm happy to collapse this to a slightly less human error message and make the test shorter if you'd prefer.

Reply:
Yeah, it's usually pretty obvious when you say `isempty=%v, wanted: %v`.

Reply:
Updated. I also noticed a bug in `URL.URL()` in the `nil` case, so I updated the tests to catch it, and then fixed the bug.